Mathlib/Order/OmegaCompletePartialOrder.lean

```diff
@@ -897,4 +899,60 @@ noncomputable def seq {β γ : Type v} (f : α →𝒄 Part (β → γ)) (g : α
 
 end ContinuousHom
 
+namespace fixedPoints
+
+open Function
+
+/-- Iteration of a function on an initial element interpreted as a chain. -/
+def iterateChain (f : α →o α) (x : α) (h : x ≤ f x) : Chain α :=
+  ⟨fun n => f^[n] x, f.monotone.monotone_iterate_of_le_map h⟩
+
+variable (f : α →𝒄 α) (x : α)
+
+/-- The supremum of iterating a function on x arbitrary often is a fixed point -/
+theorem ωSup_iterate_mem_fixedPoint (h : x ≤ f x) :
+    ωSup (iterateChain f x h) ∈ fixedPoints f := by
+  rw [mem_fixedPoints, IsFixedPt, f.continuous]
+  apply le_antisymm
+  · apply ωSup_le
+    intro n
+    simp only [Chain.map_coe, OrderHomClass.coe_coe, comp_apply]
+    have : iterateChain f x h (n.succ) = f (iterateChain f x h n) :=
+      Function.iterate_succ_apply' ..
+    rw [← this]
+    apply le_ωSup
+  · apply ωSup_le
+    rintro (_ | n)
+    · apply le_trans h
+      change ((iterateChain f x h).map f) 0 ≤ ωSup ((iterateChain f x h).map (f : α →o α))
+      apply le_ωSup
+    · have : iterateChain f x h (n.succ) = (iterateChain f x h).map f n :=
+        Function.iterate_succ_apply' ..
+      rw [this]
+      apply le_ωSup
+
+/-- The supremum of iterating a function on x arbitrary often is smaller than any prefixed point-/
+theorem ωSup_iterate_le_prefixedPoint (h : x ≤ f x) {a : α}
```

Comment:
What's `prefixedPoint` referring to?

Reply:
A prefixed Point is usually defined as a value `a` with `f a ≤ a`. In this theorem, `prefixedPoint` thus refers to the value `a` as we require that `f a ≤ a`. I am open to other names.

Reply:
I think the name is okay, but do explain it in the docstring since it doesn't refer to any Lean declaration